docker/docker_client.go

```diff
@@ -243,7 +243,13 @@ func (c *dockerClient) getBearerToken(realm, service, scope string) (string, err
 	return tokenStruct.Token, nil
 }
 
-func getAuth(hostname string) (string, string, error) {
+func getAuth(ctx *types.SystemContext, hostname string) (string, string, error) {
+	if ctx.DockerAuthConfig != nil {
+		if ctx.DockerAuthConfig.Auth != "" {
+			return decodeDockerAuth(ctx.DockerAuthConfig.Auth)
+		}
+		return ctx.DockerAuthConfig.Username, ctx.DockerAuthConfig.Password, nil
```

Comment:
Why support both `Auth` and `Username/Password` here? Do we already have real users for both which would benefit? Then we need to document which overrides what, it just seems like unnecessary API complexity.

I’d like to have only the more explicit and type-safe username/password pair, if there were no other reason to prefer the `Auth` one or to supply both.